### ISSUE TITLE
Not a issue but a misunderstanding about build options.

### ISSUE THREAD

**Issue by SirSAC** [doitsujin/dxvk#3422]
What "--dev-build" option does?
It makes the DLLs smaller?
I see in the code it adds "--strip" to Meson, which this option it should make the DLLs smaller if I remember right from how GoLang options work for his compiler.

**Comment by Blisto91** [comment]
If you call the release script with `--dev-build` it sets both `opt_nopackage` and `opt_devbuild` to `1` 
Which means we don't enter either of these if conditions since they require `0` 
```
opt_strip=
if [ $opt_devbuild -eq 0 ]; then
    opt_strip=--strip
  fi
```
```
if [ $opt_nopackage -eq 0 ]; then
  package
fi
```
In the first one we setup `opt_strip` to be empty and since we don't enter the if condition we don't set it to `--strip` right afterwards which means we won't strip the dll's and will be larger because of all the symbols. [Another if condition](https://github.com/doitsujin/dxvk/blob/65520fa18e7984b7dea9e49b4d6ee92c9a874fcf/package-release.sh#L73) further down is also skipped which means we also don't clean up our temporary build folders and files.

If we instead supply `--no-package` we don't set `opt_devbuild` to `1` and therefor enter the two if conditions associated and now tell the script we both want to strip the dll's and also clean up the temporary files and folders.

Giving the script no options does the same and in addition doesn't set `opt_nopackage` to `1` making the script enter the second if condition shown above which will go ahead and package up the final files in a archive and delete them afterwards.

**Comment by SirSAC** [comment]
Ok, is more understandable now what it does each option, thanks mate.
Have a great day and fun coding.

**Comment by doitsujin** [comment]
TL;DR the option literally only exists because I'm lazy. It's what I use to set up my build environment whenever Meson throws a fit (usually due to version incompatibilities).

**Comment by SirSAC** [comment]
You are not lazy bro, just very busy with this project.
The DXVK is so good even with some minor issues or some feature leaks, it gives me 30-40 FPS boost in all of my games, but I think it could do better with some tweaks.

**Comment by Redundanz** [comment]
what tweaks do you have in mind? where do you see untapped potential?

**Comment by SirSAC** [comment]
At this moment I'm messing around with the code until I find a modification or combination to see an improvement in performance (not placebo) but also in stability and compatibility with all my 3 PCs and each have different hardware but sadly only Intel and Nvidia, also testing with all games I have that allows me to add and load these DLLs plus special test for non-games stuff.
Of course is untapped potential, this DXVK is not even touching the 80% of Vulkan power, I'm pretty sure it can be better than DX12 and this is allowed by his open-source, community and to not forget the latest Vulkan is still 1.3 which is still young but at that version is almost the same performance as DX12, sometimes even better but others a bit lower.
Is still good amount of headroom for improvement and especially when we are talking about ray-tracing.
I will try to implement/import the "d3d9" here and maybe some changed if it doesn't work at the first run.
Also C and C++ is new for me so I'm noob, but nothing hurts if I try my own ideas on Fork.
